### PR TITLE
[ar] Deep learning

### PR DESCRIPTION
**Pull request by AmjadKhatabi**
Arabic

### CODE REVIEW

**Comment by zaidalyafeai** [comment]
@AmjadKhatabi , add [ar] tag also indicate that this is a work in progress.

**Comment by AmjadKhatabi** [comment]
[WIP] This is a work on progress!

**Comment by shervinea** [comment]
Thank you both @AmjadKhatabi and @zaidalyafeai for your help! Keep up the good work!

**Comment by AmjadKhatabi** [comment]
@shervinea I finished working on the sheet. The file is ready to be reviewed

**Comment by zaidalyafeai** [comment]
I'll review it !

**Comment by AmjadKhatabi** [comment]
Thank you zaid!

File: ar/cheatsheet-deep-learning.md [hunk 1]
```diff
@@ -1,321 +1,323 @@
...
+[Sigmoid, Tanh, ReLU, Leaky ReLU] 
+<br> 
 
 **10. Cross-entropy loss ― In the context of neural networks, the cross-entropy loss L(z,y) is commonly used and is defined as follows:**
 
 &#10230;
-
-<br>
+دالة الانتروبيا التقاطعية للخسارة - في سياق الشبكات العصبونية, دالة الاتاروبيا L(z,y) تستخدم و تعرف كالاتي:
+<br>
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: دالة الأنتروبيا

File: ar/cheatsheet-deep-learning.md [hunk 2]
```diff
@@ -1,321 +1,323 @@
...
-
-<br>
+تنظيم الحزمة - هي خطوه من قيم التحسين الخاصة γ,β  والتي تعدل الحزمة {xi}. لنجعل μB,σ2B المتوسط و الانحراف للحزمة المعنية و نريد تصحيح هذه الحزمة, يتم ذلك كالتالي:    
+<br> 
 
 **23. It is usually done after a fully connected/convolutional layer and before a non-linearity layer and aims at allowing higher learning rates and reducing the strong dependence on initialization.**
 
 &#10230;
+في الغالب تتم بعد الطبقة الالتفافية المتصلة كليا و قبل طبقة التغيرات الغير خطية و تهدف للسماح للسرعات التعليم العالية للتقليل من الاعتمادية القوية للقيم الاولية.
+
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: الطبقات الإلتفافية أو المتصلة كلياً

File: ar/cheatsheet-deep-learning.md [hunk 3]
```diff
@@ -1,321 +1,323 @@
...
+R:S×A⟶R or R:S⟶R  هي دالة المكافأة والتي تعمل الخوارزمية على جعلها اعلى قيمة
+<br> 
 
 **38. Policy ― A policy π is a function π:S⟶A that maps states to actions.**
 
 &#10230;
-
-<br>
+دالة القواعد - دالة القواعد π:S⟶A  هي التي تقوم بترجمة الاحالات الى اجراءات.
+<br>
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: الحالات

File: ar/cheatsheet-deep-learning.md [hunk 4]
```diff
@@ -1,321 +1,323 @@
...
 
 **39. Remark: we say that we execute a given policy π if given a state s we take the action a=π(s).**
 
 &#10230;
-
-<br>
-
+تعليق: نقول ان النموذج ينفذ القاعدة المعينه π للحالة المعطاة s ان نتخذ الاجراءa=π(s).  
+<br>  
+
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: Remark = ملاحظة

File: ar/cheatsheet-deep-learning.md [hunk 5]
```diff
@@ -1,321 +1,323 @@
...
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: I would prefer if you add the definition in English in parenthesis. For instance شبكة تكرارية (Recurrent Network), so people can search further about certain terms.

File: ar/cheatsheet-deep-learning.md [hunk 6]
```diff
@@ -1,321 +1,323 @@
...
+عبر تدوين i كالطبقة رقم i و j للدلالة على رقم الوحده الخفية في تلك الطبقة , نحصل على:
+<br>  
 
 **7. where we note w, b, z the weight, bias and output respectively.**
 
 &#10230;
-
-<br>
+حيث نعرف w, b, z كالوزن , و معامل التعديل , و الخرج حسب الترتيب.
+<br>
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: اعتقد يفضل استخدام كلمة ناتج مكان خرج

File: ar/cheatsheet-deep-learning.md (hunk 5 above)
Comment (zaidalyafeai) [reply ar/cheatsheet-deep-learning.md]: Add them to the definitions, RNN, CNN, Cross-Entropy, Gradient, Backpropagate, etc ...

**Comment by zaidalyafeai** [comment]
@AmjadKhatabi , initial review submitted.

**Comment by AmjadKhatabi** [comment]
initial review edit is ready..
sorry for the delay! @zaidalyafeai

File: ar/cheatsheet-deep-learning.md [hunk 7]
```diff
@@ -1,321 +1,323 @@
...
+حيث نعرف w, b, z كالوزن , و معامل التعديل , و الناتج حسب الترتيب.
+<br>  
 
 **8. Activation function ― Activation functions are used at the end of a hidden unit to introduce non-linear complexities to the model. Here are the most common ones:**
 
 &#10230;
-
-<br>
+دالة التفعيل - دالة التفعيل تستخدم في نهاية الوحده الخفية لتضمن المكونات الغير خطية للنموذج. هنا بعض دوال التفعيل الشائعة
+<br>
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: (activation function)

File: ar/cheatsheet-deep-learning.md [hunk 8]
```diff
@@ -1,321 +1,323 @@
...
-<br>
+معدل التعلم(Learning rate) - معدل التعلم, يرمز , و هو مؤشر في اي تجاة يتم تحديث الاوزان. يمكن تثبيت هذا المعامل او تحديثة بشكل تأقلمي . حاليا اكثر النسب شيوعا تدعى Adam , وهي طريقة تجعل هذه النسبة سرعة التعلم بشكل تأقلمي    α او η ب , 
+<br>  
 
 **12. Backpropagation ― Backpropagation is a method to update the weights in the neural network by taking into account the actual output and the desired output. The derivative with respect to weight w is computed using chain rule and is of the following form:**
 
 &#10230;
-
+التغذية الخلفية(Backpropagation) - التغذية الخلفية هي طريقة لتحديث الاوزان في الشبكة العصبونية عبر اعتبار القيم الحقيقة للخرج مع القيمة المطلوبة للخرج. المشتقة بالنسبة للوزن w يتم حسابها باستخدام قاعدة التسلسل و تكون عبر الشكل الاتي: 
 <br>
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: الخرج = الناتج

File: ar/cheatsheet-deep-learning.md [hunk 9]
```diff
@@ -1,321 +1,323 @@
...
+احتياج الطبقة الالتفافية - عبر رمز w لحجم المدخل , F حجم العصبونات للطبقة الالتفافية , P عدد الحشوات الصفرية , فأن N عدد العصبونات لكل حجم معطى يحسب عبر الاتي: 
 <br>
 
 **22. Batch normalization ― It is a step of hyperparameter γ,β that normalizes the batch {xi}. By noting μB,σ2B the mean and variance of that we want to correct to the batch, it is done as follows:**
 
 &#10230;
-
-<br>
+تنظيم الحزمة - هي خطوه من قيم التحسين الخاصة γ,β  والتي تعدل الحزمة {xi}. لنجعل μB,σ2B المتوسط و الانحراف للحزمة المعنية و نريد تصحيح هذه الحزمة, يتم ذلك كالتالي:    
+<br>
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: (Batch Normalization)

File: ar/cheatsheet-deep-learning.md [hunk 10]
```diff
@@ -1,321 +1,323 @@
...
+  π∗ للحالة المعطاه s تعطى كاالتالي: ملاحظة: نلاحظ ان القاعدة المثلى
+<br>  
 
 **43. Value iteration algorithm ― The value iteration algorithm is in two steps:**
 
 &#10230;
-
-<br>
+خوارزمية تكرار القيمة - خوارزمية تكرار القيمة تكون في خطوتين:
+<br>
```
Comment (zaidalyafeai) [inline ar/cheatsheet-deep-learning.md]: Value Iteration algorithm

**Comment by zaidalyafeai** [comment]
@AmjadKhatabi , check my final suggestions! Great work!!

**Comment by AmjadKhatabi** [comment]
Hello @zaidalyafeai , any other suggestions?!

**Comment by zaidalyafeai** [comment]
@shervinea review phase is finished.

**Comment by shervinea** [comment]
Thanks everyone for your work! Now let's be sure that everyone is added to the `CONTRIBUTORS` file before merging.

**Comment by zaidalyafeai** [comment]
@AmjadKhatabi, add your name and mine to the contributers file.

**Comment by AmjadKhatabi** [comment]
All set!

**Comment by shervinea** [comment]
Thanks @AmjadKhatabi and @zaidalyafeai! It looks like we are all set now!
